### PR TITLE
[xh]Add test functions for AI class

### PR DESCRIPTION
**Pull request by matrixstone**
# Description
Add 5 tests for all public methods in llm_pipeline_wizard

# How Has This Been Tested?
`python -m unittest mage_ai/tests/ai/test_ai_functions.py`

# Checklist
- [X] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [X] I have performed a self-review of my own code
- [X] I have added unit tests that prove my fix is effective or that my feature works

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_ai/tests/ai/test_ai_functions.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                         mock_customized_code,
+                         mock_customized_code,
+                         mock_customized_code])
+        self.wizard._LLMPipelineWizard__async_identify_function_parameters = MagicMock(
+            side_effect=[
+                self.__set_expected_future_result(MOCK_LOADER_FUNCTION_CLASSIFICATION),
+                self.__set_expected_future_result(MOCK_TRANSFORMER_CLASSIFICATION),
+                self.__set_expected_future_result(MOCK_EXPORTERS_FUNCTION_CLASSIFICATION)])
+
+        pipeine_description = "load data from mysql, filter out "
```
Comment (wangxiaoyou1993) [inline mage_ai/tests/ai/test_ai_functions.py:77]: for multiline string, you'll need to put `\` at the end of the line
a = "aaaaa" \ 
  "bbbbb"

File: mage_ai/tests/ai/test_ai_functions.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+EXPECTED_RESPONSE_FOR_LLM_SPLIT = """
+Answer:
+BLOCK 1: function: load data from MySQL. upstream:
+BLOCK 2: function: filter out records column age < 18. upstream: 1
+BLOCK 3: function: export it to postgres. upstream: 2"""
+EXPECTED_LOAD_BLOCK = '{"code": "load mysql"}'
+EXPECTED_FILTER_BLOCK = '{"code": "filter record"}'
+EXPECTED_EXPORT_BLOCK = '{"code": "export postgres"}'
+MOCK_TRANSFORMER_CLASSIFICATION = {
+    "role": "assistant",
```
Comment (wangxiaoyou1993) [inline mage_ai/tests/ai/test_ai_functions.py:18]: nit: we prefer single quote in the python code.

Comment (matrixstone) [reply mage_ai/tests/ai/test_ai_functions.py:18]: Done

File: mage_ai/tests/ai/test_ai_functions.py (hunk 1 above)
Comment (matrixstone) [reply mage_ai/tests/ai/test_ai_functions.py]: Updated

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -68,7 +68,12 @@ jobs:
           pip install "git+https://github.com/mage-ai/singer-python.git#egg=singer-python"
       - name: Run unit tests
         run: |
-          python3 -m unittest discover -s mage_ai --failfast
+          python3 -m unittest discover -s mage_ai --failfast -p "^(?!.*test_ai_function).*$"
+      - name: Run AI unit tests
```
Comment (wangxiaoyou1993) [inline .github/workflows/build_and_test.yml:72]: No need to add this. You can rebase master. We exclude 3.7 in our unit tests.

Comment (matrixstone) [reply .github/workflows/build_and_test.yml]: Sounds good! Reverted the build_and_test yaml change.

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**